pygmt/clib/session.py

```diff
@@ -2311,3 +2312,29 @@ def extract_region(self) -> np.ndarray:
         if status != 0:
             raise GMTCLibError("Failed to extract region from current figure.")
         return region
+
+
+def _raster_kind(raster: str) -> Literal["grid", "image"] | None:
```

Comment:
Mypy is complaining about the output type hint here:

```
pygmt/src/grdcut.py:108: error: Incompatible types in assignment (expression has type "Literal['grid', 'image'] | None", variable has type "Literal['grid', 'image']")  [assignment]
```

Maybe we should just use `Literal['grid', 'image']` and raise `GMTInvalidInput` if it is not a grid/image?

Reply:
Maybe falling back to `"grid"` and then let GMT itself decide if it can process it or not.

Reply:
In that case, with a binary image/grid output, maybe just make a function like `def _is_image() -> bool`?

Reply:
grdcut can also cut a cube, so the `_raster_kind` function should return `grid`/`image`/`cube` if possible. Currently, the' cube' support is not yet finished (PR #3150). So, the name `_raster_kind` makes more sense.

Reply:
Ah yes, and then there would be 3 try-except statements. Now wishing we could re-use the function from GMT C...